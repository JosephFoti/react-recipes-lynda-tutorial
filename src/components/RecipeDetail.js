import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const RecipeDetail = (props) => {
  if (!props.recipe) {
    return (
      <p
        style={props.style}
        className={classNames('h3 p2 bg-white italic center', props.className)}
      >
        Please select a recipe to see the detail
      </p>
    );
  }

  return (

    <div
      style={props.style}
      className={classNames('p2 bg-white', props.className)}
    >
      <h2 className="h2">
        {props.recipe.name}
      </h2>
      <img className="fit" src={props.recipe.image} alt={props.recipe.name} />
      <div>
        <span>
          {props.recipe.category}
        </span>
        <span>
          {props.recipe.calories}
        </span>
      </div>
      <h3>
        Ingredients
      </h3>

      <ul>

        {props.recipe.ingredients.map(ingredient => (
          <li
            key={ingredient}
          >
            {ingredient}
          </li>
        ))}

      </ul>

      <h3>
        Steps
      </h3>

      <ol>

        {props.recipe.steps.map(step => (
          <li key={step}>
            {step}
          </li>
        ))}
      </ol>


    </div>

  );
};

RecipeDetail.propTypes = {
  recipe: PropTypes.object,
  className: PropTypes.string,
  style: PropTypes.object,
};

export default RecipeDetail;
