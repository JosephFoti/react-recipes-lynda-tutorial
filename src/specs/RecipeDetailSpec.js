import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';

describe('<RecipeDetail />', () => {
  let testRecipe;

  beforeEach(() => {
    testRecipe = {
      id: 1,
      name: 'Test Recipe',
      category: 'Test category',
      ingredients: ['ing1', 'ing2'],
      steps: ['stp1', 'stp2'],
    };
  });

  test('Should render zero state', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render recipe', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should assign ClassName', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} className="classname" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should recipe without ingredients', () => {
    delete testRecipe.ingredients;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should recipe without steps', () => {
    delete testRecipe.steps;
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={testRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
