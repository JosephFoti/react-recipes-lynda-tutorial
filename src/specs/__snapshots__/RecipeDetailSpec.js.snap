// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> Should assign ClassName 1`] = `
<div
  className="p2 bg-white classname"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      ing1
    </li>
    <li>
      ing2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      stp1
    </li>
    <li>
      stp2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should recipe without ingredients 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      stp1
    </li>
    <li>
      stp2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should recipe without steps 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      ing1
    </li>
    <li>
      ing2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should render recipe 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    Test Recipe
  </h2>
  <img
    alt="Test Recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      ing1
    </li>
    <li>
      ing2
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      stp1
    </li>
    <li>
      stp2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See More
  </a>
</div>
`;

exports[`<RecipeDetail /> Should render zero state 1`] = `
<p
  className="h3 p2 bg-white italic center"
>
  Please select a recipe to see the detail
</p>
`;
